ELMOD1.py

```diff
@@ -0,0 +1,317 @@
+
+    def __len__(self) -> int:
+        """
+        Return the length of the sequence.
+
+        Returns:
+        - int: Length of the sequence.
+        """
+        return len(self.sequence)
+    
+    def __getitem__(self, index: int) -> str:
+        """
+        Get the character at the specified index in the sequence.
+
+        Parameters:
+        - index (int): The index to retrieve.
+
+        Returns:
+        - str: The character at the specified index.
+        """
+        return self.sequence[index]
+    
+    def __str__(self) -> str:
+        """
+        Return the string representation of the sequence.
+
+        Returns:
+        - str: String representation of the sequence.
+        """
+        return self.sequence
+    
+    @abstractmethod
+    def is_valid_alphabet(self) -> bool:
+        """
+        Check if the sequence contains a valid alphabet.
+
+        Returns:
+        - bool: True if the alphabet is valid, False otherwise.
+        """
+        pass
+
+    def validate_alphabet(self) -> None:
+        """
+        Validate the alphabet of the sequence.
+
+        Raises:
+        - ValueError: If the alphabet is not valid.
+        """
+        if not self.is_valid_alphabet():
+            raise ValueError("Invalid alphabet for {}: {}".format(self.__class__.__name__, str(self)))
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Class representing nucleic acid sequences.
+    """
+    def complement(self) -> 'NucleicAcidSequence':
+        """
+        Return the complement of the sequence.
+
+        Returns:
+        - NucleicAcidSequence: Complement of the sequence.
+        """
+        self.validate_alphabet()
+        return self.__class__(''.join([self.complement_base(base) for base in self.sequence]))
+    
+    def gc_content(self) -> float:
+        """
+        Calculate the GC content of the sequence.
+
+        Returns:
+        - float: GC content of the sequence.
+        """
+        self.validate_alphabet()
+        gc_count = sum(1 for base in self.sequence if base in {'G', 'C'})
+        return gc_count / len(self)
+
+class DNASequence(NucleicAcidSequence):
+    """
+    Class representing DNA sequences.
+    """
+    def transcribe(self) -> 'RNASequence':
+        """
+        Transcribe the DNA sequence into RNA.
+
+        Returns:
+        - RNASequence: Transcribed RNA sequence.
+        """
+        self.validate_alphabet()
+        return RNASequence(''.join(['U' if base == 'T' else base for base in self.sequence]))
+    
+    def complement_base(self, base: str) -> str:
+        """
+        Return the complement of a DNA base.
+
+        Parameters:
+        - base (str): The DNA base.
+
+        Returns:
+        - str: Complement of the DNA base.
+        """
+        if base == 'A':
+            return 'T'
+        elif base == 'T':
+            return 'A'
+        elif base == 'C':
+            return 'G'
+        elif base == 'G':
+            return 'C'
```

Comment:
Мне кажется, что логичнее и правильнее реализовать это через словарь:

```suggestion
complement_dict = {'A': 'T', 'T': 'A', 'C': 'G', 'G': 'C'}
        return complement_dict.get(base)
```